app/models/user_system_notification.rb

```diff
@@ -8,4 +8,42 @@ def self.mark_as_seen(notifications)
     where(system_notification_id: notifications)
       .update_all(seen_at: Time.now)
   end
+
+  def self.mark_read(notification_id)
+    notification = find_by_system_notification_id(notification_id)
+    if notification && notification.read_at.nil?
+      notification.update(read_at: Time.now)
+    end
+  end
+
+  def self.modal(notification_id)
+    select(:modal_title, :modal_body, 'system_notifications.id')
+      .joins(:system_notification)
+      .find_by_system_notification_id(notification_id)
+  end
+
+  def self.show_on_login(update_read_time = false)
```

Comment:
So we are flagging as read as soon as we are returning it. Shouldn't we `mark_as_read` with reusing `mark_read` method (or maybe service)?

Reply:
If we reuse mark_read method, than we will need perform 2 updates - 1 for update seen_at and 1 for read_at.

Reply:
Yes, I know. I was thinking to mark as_read with ajax when user get a page. Probably overkill for this case...